### PR TITLE
Retrying on tenant startup

### PR DESCRIPTION
**Pull request by sebastienros**
Targeting 1.10 to focus on the changes. Don't merge.

### CODE REVIEW

File: src/Orchard/Environment/DefaultOrchardHost.cs [hunk 1]
```diff
@@ -142,16 +144,25 @@ public class DefaultOrchardHost : IOrchardHost, IShellSettingsManagerEventHandle
...
+                    for (var i = 0; i <= Retries; i++) {
+                        bool failed = false;
+                        try {
+                            var context = CreateShellContext(settings);
+                            ActivateShell(context);
+                        }
+                        catch (Exception ex) {
+                            // An exception at this point is always fatal as it literally kills the
+                            // tenant. What is more fatal than something that kills you?
+                            Logger.Error(ex, "A tenant could not be started: " + settings.Name + " Try number: " + i);
```
Comment (sebastienros) [inline src/Orchard/Environment/DefaultOrchardHost.cs]: I don't like the word 'Try', 'Tentative' seems better, any other suggestion?

**Comment by sebastienros** [comment]
The scenario is that at startup there might be some SQL hiccups and the tenant would not start, and you would not know unless you read the error logs frequently. As Azure Websites  restarts the sites regularly, some sites might go down. It happens sometimes on weblogs.asp.net as we have 800 tenants.

**Comment by dcinzona** [comment]
I like this change.  This can happen for us as well.  What is the retry timeout? Exponential?

File: src/Orchard/Environment/DefaultOrchardHost.cs (hunk 1 above)
Comment (dcinzona) [inline src/Orchard/Environment/DefaultOrchardHost.cs]: Attempt?

**Comment by sebastienros** [comment]
What do people think about retry delay?

**Pull request closed without merging**